### PR TITLE
Added support for STM L3G4200D digital gyroscope

### PR DESCRIPTION
**Pull request by jschmidlapp**

### CODE REVIEW

**Comment by marcelstoer** [comment]
Can't comment on the C code but on a number of formalities. See https://github.com/nodemcu/nodemcu-firmware/blob/dev/CONTRIBUTING.md#writing-documentation:

- we appreciate descriptions for PRs
- modules in `user_modules.h` should be alphabetically sorted a-z
- module documentation should be added to `mkdocs.yml`
- modules in `mkdocs.yml` should be alphabetically sorted a-z
- functions in the module documentation should be alphabetically sorted a-z

File: app/modules/l3g4200d.c [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    platform_i2c_send_stop(i2c_id);
+}
+
+static int ICACHE_FLASH_ATTR l3g4200d_init(lua_State* L) {
+
+    uint32_t sda;
+    uint32_t scl;
+    uint8_t  devid;
+
+    if (!lua_isnumber(L, 1) || !lua_isnumber(L, 2)) {
```
Comment (TerryE) [inline app/modules/l3g4200d.c]: These test is redundant as the checkinteger tests do this anyway.

File: app/modules/l3g4200d.c [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        return luaL_error(L, "wrong arg range");
+    }
+
+    sda = luaL_checkinteger(L, 1);
+    scl = luaL_checkinteger(L, 2);
+
+    if (scl == 0 || sda == 0) {
```
Comment (TerryE) [inline app/modules/l3g4200d.c]: Use a luaL_argcheck() for these tests.  The code is shorter and the error message more informative.

File: app/modules/l3g4200d.c [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    platform_i2c_send_address(i2c_id, i2c_addr, PLATFORM_I2C_DIRECTION_TRANSMITTER);
+    platform_i2c_send_byte(i2c_id, 0xA8);
+    platform_i2c_send_start(i2c_id);
+    platform_i2c_send_address(i2c_id, i2c_addr, PLATFORM_I2C_DIRECTION_RECEIVER);
+
+    for (i=0; i<5; i++) {
+	data[i] = platform_i2c_recv_byte(i2c_id, 1);
+    }
+    
+    data[5] = platform_i2c_recv_byte(i2c_id, 0);
```
Comment (TerryE) [inline app/modules/l3g4200d.c]: Why unroll this one?

**Comment by jmattsson** [comment]
:+1: with similar comments to #1252. Then again, I guess across all of these there's a convention emerging... :D

**Pull request merged**